scripts/release_scripts/cut_release_or_hotfix_branch.py

```diff
@@ -296,8 +296,9 @@ def execute_branch_cut(target_version, hotfix_number):
         subprocess.check_call(['git', 'checkout', '-b', new_branch_name])
 
     # Push the new release branch to GitHub.
-    python_utils.PRINT('Pushing new %s branch to GitHub.' % new_branch_type)
-    subprocess.check_call(['git', 'push', remote_alias, new_branch_name])
+    if new_branch_type == release_constants.BRANCH_TYPE_RELEASE:
+        python_utils.PRINT('Pushing new %s branch to GitHub.' % new_branch_type)
+        subprocess.check_call(['git', 'push', remote_alias, new_branch_name])
```

Comment:
We do not need to push the branch before adding a protection rule, i.e. branch does not need to exist for the rule to be defined.

I have updated the script to print a message to remind user to cherrypick PRs. I do not think user should do it in a new terminal as it might create confusion if cherrypicks aren't simple and there are merge conflicts. WDYT?

Reply:
> We do not need to push the branch before adding a protection rule, i.e. branch does not need to exist for the rule to be defined.

I think you define protection to that specific branch, right? And for that the branch need to exist, right?

> I do not think user should do it in a new terminal as it might create confusion if cherrypicks aren't simple and there are merge conflicts. WDYT?

Fixing merge conflicts will not create any issue for the script, right? If no, then I was thinking of this script to complete the whole hot-fix branch cut to push process. How do we push cherry-picked PR in the release process is it manually or via a script, if via a script then I think we should complete the cut, pick and push process here maybe? wdyt?

Reply:
> I think you define protection to that specific branch, right? And for that the branch need to exist, right?

We define protection like a regex i.e. it will be followed for all branches that follow that name pattern. So, if a rule is in place and a branch is pushed later as well, the rule will be followed.

> Fixing merge conflicts will not create any issue for the script, right? If no, then I was thinking of this script to complete the whole hot-fix branch cut to push process. How do we push cherry-picked PR in the release process is it manually or via a script, if via a script then I think we should complete the cut, pick and push process here maybe? wdyt?

We complete the cherrypick process manually in the release process since it requires user intervention specifically with merge conflicts and I feel a script running in parallel can be confusing.

Reply:
Thanks for all the info, sg for all!